src/vs/editor/contrib/stickyScroll/browser/stickyScrollProvider.ts

```diff
@@ -92,7 +94,21 @@ export class StickyLineCandidateProvider extends Disposable {
 			if (token.isCancellationRequested) {
 				return;
 			}
-			this._outlineModel = StickyOutlineElement.fromOutlineModel(outlineModel);
+			if (outlineModel.children.size !== 0) {
+				this._outlineModel = StickyOutlineElement.fromOutlineModel(outlineModel);
+			} else {
+				const foldingController = FoldingController.get(this._editor);
+				const foldingModel = await foldingController?.getFoldingModel();
+				if (foldingModel) {
```

Comment:
You could check the folding model for being empty here (`foldingModel.regions.length`) and only continue if not. That allows you to drop lines 211-217 which are similar/same to 105-109